moveit_ros/planning/rdf_loader/test/test_rdf_integration.cpp

```diff
@@ -66,6 +66,30 @@ TEST(RDFIntegration, topic_based)
   EXPECT_EQ("gonzo", loader.getSRDF()->getName());
 }
 
+// RDFLoader should successfully load URDF and SRDF strings from a ROS topic when the node that is passed to it is
+// spinning.
+// GIVEN a node that has been added to an executor that is spinning on another thread
+// WHEN the RDFLoader is created
+// THEN the RDFLoader should return non-null values for the URDF and SRDF model.
+TEST(RDFIntegration, executor)
+{
+  rclcpp::Node::SharedPtr node = std::make_shared<rclcpp::Node>("executor");
+
+  // Create a thread to spin an Executor.
+  std::thread([node]() {
+    rclcpp::executors::SingleThreadedExecutor executor;
+    executor.add_node(node);
+    executor.spin();
+  }).detach();
+
+  rdf_loader::RDFLoader loader(node, "topic_description");
+
+  ASSERT_NE(nullptr, loader.getURDF());
+  EXPECT_EQ("gonzo", loader.getURDF()->name_);
+  ASSERT_NE(nullptr, loader.getSRDF());
+  EXPECT_EQ("gonzo", loader.getSRDF()->getName());
+}
```

Comment:
```suggestion
TEST(RDFIntegration, executor)
{
  // RDFLoader should successfully load URDF and SRDF strings from a ROS topic when the node that is 
  // passed to it is spinning.
  // GIVEN a node that has been added to an executor that is spinning on another thread
  rclcpp::Node::SharedPtr node = std::make_shared<rclcpp::Node>("executor");

  // Create a thread to spin an Executor.
  std::thread([node]() {
    rclcpp::executors::SingleThreadedExecutor executor;
    executor.add_node(node);
    executor.spin();
  }).detach();

  // WHEN the RDFLoader is created
  rdf_loader::RDFLoader loader(node, "topic_description");

  // THEN the RDFLoader should return non-null values for the URDF and SRDF model.
  ASSERT_NE(nullptr, loader.getURDF());
  EXPECT_EQ("gonzo", loader.getURDF()->name_);
  ASSERT_NE(nullptr, loader.getSRDF());
  EXPECT_EQ("gonzo", loader.getSRDF()->getName());
}
```

Reply:
note that I did not run the formatted on this... I don't know if I got the line lengths right.  I was just trying to make it more clear where I would like the comments.